### PR TITLE
Define the "movement dataset" and reorganise the Getting Started section

### PR DESCRIPTION
**Pull request by niksirbi**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

The recently merged PR #174 changed the way we conceptualise and use the `xarray.Dataset` accessor, aligning it closer with this [proposal](https://github.com/neuroinformatics-unit/movement/issues/162). This prompted me to extensively document the current structure and usage of this accessor, as was described in the proposal.

**What does this PR do?**

I renamed the `MoveAccessor` to `MovementDataset`, which may be more transparent to users. the module is still named `move_accessor.py`, but the `MovementDataset` class can be directly imported as `from movement import MovementDataset` (the user doesn't need to know about the `move_accessor.py` module). I significantly shrunk the docsrtring of the `MovementDataset` class, because we will now have a whole docs section to serve as the authoritative source for how the dataset is structured (see next point).

I vastly expanded the description of dataset structure, which is now a standalone page named "The movement dataset". Within it, I describe the structure and usage of movement datasets, built-in vs movement-specific functionalities etc.

Since the above section was "split away" from the Getting Started guide, I decided to also refactor the rest of the guide into more modular sections. This is not yet fully following #70 but is hopefully a step in the right direction. The Getting Started section now has the following pages/subsections:
- installation (self-explanatory, this includes the `movement info` command for checking the installation
- input/output (supported file formats, loading and saving files)
- sample data (which was previously "hidden" in a dropdown, this is up-to-date with https://github.com/neuroinformatics-unit/movement/pull/171)
- the movement dataset (as described above).

Keep in mind that this is not the endpoint for how the docs will look like. More restructuring is to follow. That said, the updated Getting Started section should serve its purpose for the upcoming Measuring Behavior conference, and I needed to make sure that it reflects recent changes in the accessor and in sample data.

This is how the Getting Started section looks with these changes:
Screenshot from 2024-05-13 18-33-41

## How to review
The best way is to build the docs locally and inspect the rendered result. You may ignore syntax such as `class{xarray.Dataset}` (in must markdown) and ``:py:class:`xarray.Dataset` `` (in docstrings and rst). They don't have much of an effect right now, but they will pay off once we configure intersphinx.

## References

https://github.com/neuroinformatics-unit/movement/issues/162
https://github.com/neuroinformatics-unit/movement/issues/70
https://github.com/neuroinformatics-unit/movement/pull/171
https://github.com/neuroinformatics-unit/movement/pull/174

## How has this PR been tested?

Please explain how any new code has been tested, and how you have ensured that no existing functionality has changed.

## Is this a breaking change?

No

## Does this PR require an update to the documentation?

It's all about documentation.

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: docs/source/getting_started/movement_dataset.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+the labelled "ticks" along each axis are called **coordinates** (`coords`).
+
+A `movement` dataset has the following **dimensions**:
+- `time`, with size equal to the number of frames in the video
+- `individuals`, with size equal to the number of tracked individuals/instances
+- `keypoints`, with size equal to the number of tracked keypoints per individual
+- `space`: the number of spatial dimensions, either two (2D) or three (3D)
+
+Appropriate **coordinates** are assigned to each **dimension**.
+- `individuals` are labelled with a list of unique names (str), e.g. `mouse1`, `mouse2`, etc.
```
Comment (lochhh) [inline docs/source/getting_started/movement_dataset.md:46]:
```suggestion
- `individuals` are labelled with a list of unique names, e.g. `mouse1`, `mouse2`, etc.
```

File: docs/source/getting_started/movement_dataset.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+- `keypoints` are likewise labelled with a list of unique body part names (str), e.g. `snout`, `right_ear`, etc.
```
Comment (lochhh) [inline docs/source/getting_started/movement_dataset.md:47]:
```suggestion
- `keypoints` are likewise labelled with a list of unique body part names, e.g. `snout`, `right_ear`, etc.
```

File: docs/source/getting_started/movement_dataset.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+### Attributes
+
+Related metadata that do not constitute arrays but instead take the form
+of key-value pairs can be stored as **attributes** - i.e. inside the specially
+designated `attrs` dictionary. Right after loading a `movement` dataset,
+the following **attributes** are stored:
+- `fps`: the number of frames per second in the video
+- `time_unit`: the unit of the `time` **coordinates**, frames or seconds
+- `source_software`: the software from which the poses were loaded
```
Comment (lochhh) [inline docs/source/getting_started/movement_dataset.md:69]:
```suggestion
- `source_software`: the software from which the poses were output
```
?

File: docs/source/getting_started/index.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+Alternatively, you can use the [sample data](target-sample-data)
+provided with the package.
+
+Once you have loaded some data, feel free to explore the structure and
+usage of [movement datasets](movement_dataset.md), which are a central
+concept in the package.
+
+```{toctree}
+:maxdepth: 2
+:caption: Contents:
```
Comment (lochhh) [inline docs/source/getting_started/index.md:16]:
```suggestion
:caption: Contents
```
The : is giving a bad yaml warning

File: docs/source/api_index.rst [hunk 5]
```diff
@@ -71,13 +71,13 @@ Analysis
     cart2pol
     pol2cart
 
-Move Accessor
--------------
+MovementDataset
```
Comment (lochhh) [inline docs/source/api_index.rst:74]: Something is off with the auto api, as it doesn't show all methods

**Review by lochhh — CHANGES_REQUESTED** [review]
Great work (also amount) @niksirbi! As usual, comments and suggestions are minor/aesthetic.

File: docs/source/getting_started/input_output.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+from movement.io import load_poses
+```
+
+Depending on the source sofrware, one of the following functions can be used.
+
+::::{tab-set}
+
+:::{tab-item} SLEAP
+
+Load from [SLEAP analysis .h5 files](sleap:tutorials/analysis):
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply docs/source/getting_started/input_output.md]: Good point, I added a note that you can do that, but with caveats.

```md
You can also load from SLEAP .slp files in the same way, but there are caveats
to that approach (see notes in {func}`movement.io.load_poses.from_sleap_file`).
```

File: docs/source/getting_started/movement_dataset.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+printing it:
+```python
+ds = load_poses.from_dlc_file("/path/to/file.h5", fps=30)
+print(ds)
+```
+If you are working in a Jupyter notebook, you can also view an interactive
+representation of the dataset by typing its name - e.g. `ds` - in a cell.
+
+### Dimensions and coordinates
+In `xarray` [terminology](xarray:user-guide/plotting.html),
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply docs/source/getting_started/movement_dataset.md:35]: good catch!

File: docs/source/getting_started/input_output.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+    "/path/to/file.analysis.csv", source_software="LightningPose", fps=30
+)
+```
+:::
+
+::::
+
+The loaded data include the predicted positions for each individual and
+keypoint as well as the associated point-wise confidence values, as reported by
+the pose estimation software. See the [movement dataset](target-dataset) page
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply docs/source/getting_started/input_output.md:87]: I don't like how the theme renders the combination of monospace and hyperlink (the visual result looks a bit confusing). So I've decided to use "`movement` dataset", unless it's in a hyperlink, in which case I use plain "movement dataset".

File: docs/source/getting_started/sample_data.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+that was used to generate it - either "DLC", "SLEAP", or "LP".
+
+To load one of the sample datasets into `movement`, you can use the
+{func}`movement.sample_data.fetch_dataset()` function:
+
+```python
+ds = sample_data.fetch_dataset("DLC_two-mice.predictions.csv")
+```
+This function loads the sample pose data into `movement` as a
+[movement dataset](target-dataset). Some sample datasets may also have an
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply docs/source/getting_started/sample_data.md:28]: see my response to the other similar comment.

File: docs/source/api_index.rst (hunk 5 above)
Comment (niksirbi) [reply docs/source/api_index.rst:74]: Issue opened here: https://github.com/neuroinformatics-unit/movement/issues/178

**Review by lochhh — APPROVED** [review]

**Pull request merged**